skimage/segmentation/_watershed_cy.pyx

```diff
@@ -123,7 +127,10 @@ def watershed_raveled(cnp.float64_t[::1] image,
     with nogil:
         for i in range(marker_locations.shape[0]):
             index = marker_locations[i]
-            elem.value = image[index]
+            if from_minima:
+                elem.value = image[index]
+            else:
+                elem.value = -2.2e-308  # almost the minimum value of float64
```

Comment:
This value is extremely close to 0 so it is unlikely to be the smallest float64 value. 😂 

```
In [1]: np.finfo(np.float64)
Out[1]: finfo(resolution=1e-15, min=-1.7976931348623157e+308, max=1.7976931348623157e+308, dtype=float64)
```

I suggest finding the value outside of the nogil block, `min_float64 = np.finfo(np.float64).min` and using `min_float64` here